File: easy/problems/0482-license-key-formatting/solution.py
def licenseKeyFormatting(S, K):
    """
    Format a license key with dashes.

    Args:
        S: str - license key string
        K: int - characters between dashes

    Returns:
        str - formatted license key
    """
    cleaned = S.replace('-', '').upper()

    if not cleaned:
        return ""

    result = []
    for i in range(len(cleaned) - 1, -1, -1):
        if len(result) > 0 and len(result) % (K + 1) == K:
            result.append('-')
        result.append(cleaned[i])

    return ''.join(reversed(result))

File: easy/problems/0482-license-key-formatting/solution.ts
function licenseKeyFormatting(S: string, K: number): string {
    const cleaned = S.replace(/-/g, '').toUpperCase();

    if (cleaned.length === 0) return "";

    const result: string[] = [];
    for (let i = cleaned.length - 1; i >= 0; i--) {
        if (result.length > 0 && result.length % (K + 1) === K) {
            result.push('-');
        }
        result.push(cleaned[i]);
    }

    return result.reverse().join('');
}
